src/extras/named-register.js

```diff
@@ -40,4 +39,14 @@
   systemJSPrototype.instantiate = function (url, firstParentUrl) {
     return this.registerRegistry[url] || instantiate.call(this, url, firstParentUrl);
   };
+
+  const getRegister = systemJSPrototype.getRegister;
+  systemJSPrototype.getRegister = function () {
+    let result = getRegister.apply(this, arguments);
```

Comment:
I'm really not keen on the `isEmpty` property, if it is possible to avoid. We might be able to make this - 

```js
if (lastNamedDefine) {
  lastNamedDefine = null;
  return lastNamedDefine;
}
return getRegister.apply(this, arguments);
```

The above would work provided we can guarantee that `lastNamedDefine` is **only** defined when we are within the module body of a named register module, which seems valid.

The only exception to that might be calling `System.register('asdf', ...)` not as part of a bundle module execution.

One way to handle that would be to rely on the fact that getRegister must be called synchronously after the System.register for the "bundle" cases.

So System.register could do something like `setImmediate(() => { lastNamedDefine = null })` just after setting `lastNamedDefine`, which would invalidate those other cases.